### PR TITLE
Richmond dev

### PR DESCRIPTION
**Pull request by chmnata**
## What this pull request accomplishes:
Get radioitems that looks change style when selected and other small style changes into heroku

`Screenshot of dashboard before`
image

`Screenshot of dashboard after`
image

## Issue(s) this solves:
- #46 
- #43 
- #42 
- #29 
- #35 
- #41 
- #40 

## What, in particular, needs to reviewed:
- css changes
- dbc structure
- addition of sql file

### CODE REVIEW

**Comment by radumas** [comment]
are the tabs supposed to not take up the whole window?
image

File: assets/dashboard.css [hunk 1]
```diff
@@ -73,8 +74,8 @@ td.same {
    font-size: 1.4rem; }
 
 .twelve.columns { background-color: #165788; }
-h1 { font-size: 2.0rem; line-height: 1.2; letter-spacing: -.1rem; margin-top: 0.6rem; margin-bottom: 0.6rem; margin-left: 0.6rem; color: #FFFFFF; font-weight: bold; }
-h2 { font-size: 1.6rem; line-height: 1.1; letter-spacing: -.1rem; margin-bottom: 0.4rem; margin-top: 0.4rem; margin-left: 0.6rem;}
+h1 { font-size: 2.0rem; line-height: 1.2; letter-spacing: -.1rem; margin-top: 0.6rem; margin-bottom: 0.6rem; margin-left: 0.6rem; color: #FFFFFF; font-weight: bold; background-color: #165788;}
```
Comment (radumas) [inline assets/dashboard.css]: Why does H1 have a background-color?

File: assets/dashboard.css [hunk 2]
```diff
@@ -220,7 +221,7 @@ body {
   font-size: 1.5em; /* currently ems cause chrome bug misinterpreting rems on body element */
   line-height: 1.6;
   font-weight: 400;
-  font-family: "Open Sans", "HelveticaNeue", "Helvetica Neue", Helvetica, Arial, sans-serif;
+  font-family: Arial;
```
Comment (radumas) [inline assets/dashboard.css]: Is it possible to use "Libre Franklin" instead?

File: sql/baseline_view.sql [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                    WHEN a.street = 'Front'::text AND a.direction = 'WB'::text THEN 'Yonge'::text
+                    WHEN (a.street = ANY (ARRAY['Bathurst'::text, 'Spadina'::text, 'University'::text])) AND a.direction = 'SB'::text THEN 'Dundas'::text
+                    ELSE NULL::text
+                END AS from_intersection,
+                CASE
+                    WHEN (a.street = ANY (ARRAY['Adelaide'::text, 'Front'::text, 'King'::text, 'Adelaide'::text, 'Richmond'::text, 'Queen'::text, 'Dundas'::text])) AND a.direction = 'EB'::text THEN 'Yonge'::text
+                    WHEN (a.street = ANY (ARRAY['Bathurst'::text, 'Spadina'::text, 'University'::text])) AND a.direction = 'NB'::text THEN 'Dundas'::text
+                    WHEN (a.street = ANY (ARRAY['Adelaide'::text, 'Front'::text, 'King'::text, 'Adelaide'::text, 'Richmond'::text, 'Queen'::text, 'Dundas'::text])) AND a.direction = 'WB'::text THEN 'Bathurst'::text
+                    WHEN (a.street = ANY (ARRAY['Bathurst'::text, 'Spadina'::text, 'University'::text])) AND a.direction = 'SB'::text THEN 'Front'::text
```
Comment (radumas) [inline sql/baseline_view.sql]: This kind of logic sequence is a bit of a code smell. I feel like the from-to intersections should be in a lookup table.

File: app.py [hunk 4]
```diff
@@ -40,7 +41,7 @@
                         date, day_type, category, period, tt, most_recent, week_number, month_number from data_analysis.richmond_dash_daily
                          ''', con)
 BASELINE = pandasql.read_sql('''select street, street_suffix, direction, from_intersection, to_intersection, 
-                             day_type, period, to_char(lower(period_range::TIMERANGE), 'FMHH A.M.')||' to '||to_char(upper(period_range::TIMERANGE), 'FMHH A.M.') as period_range , tt
+                             day_type, period, to_char(lower(period_range::TIMERANGE), 'FMHH AM')||' to '||to_char(upper(period_range::TIMERANGE), 'FMHH AM') as period_range , tt
```
Comment (radumas) [inline app.py:44]: Nice

File: app.py [hunk 5]
```diff
@@ -49,12 +50,12 @@
 HOLIDAY = pandasql.read_sql(''' SELECT dt FROM ref.holiday WHERE dt > '2019-07-02' ''', con, parse_dates=['dt',])
 
 # Numbering Weeks and Months for Dropdown Selectors
-WEEKS = pandasql.read_sql('''SELECT * FROM data_analysis.richmond_closure_weeks 
+WEEKS = pandasql.read_sql('''SELECT * FROM data_analysis.richmond_closure_weeks order by week_number desc
```
Comment (radumas) [inline app.py:53]: Does anything need to be changed in the data in the heroku postgresql database?

File: app.py [hunk 6]
```diff
@@ -103,7 +104,7 @@
                    baseline='#7f7e7e',
                    selected='#ec9f09'
                    )
-FONT_FAMILY = '"Open Sans", "HelveticaNeue", "Helvetica Neue", Helvetica, Arial, sans-serif'
+FONT_FAMILY = 'Arial'
```
Comment (radumas) [inline app.py]: Libre Franklin?

File: app.py [hunk 7]
```diff
@@ -143,7 +146,7 @@
 metas = [{'name':"viewport",
          'content':"width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no"}]
 
-app = dash.Dash(__name__, meta_tags=metas)
+app = dash.Dash(__name__, meta_tags=metas, external_stylesheets=[dbc.themes.BOOTSTRAP])
```
Comment (radumas) [inline app.py:149]: Sweet!
Do you think there should be a write-up somewhere about how we switched these components to dbc?

File: app.py [hunk 8]
```diff
@@ -532,7 +533,8 @@ def generate_figure(street, direction, day_type='Weekday', period='AMPK',
                   xaxis=dict(title='Date',
                              tickformat = '%b %d',
                              nticks = tick_number,
-                             fixedrange=True), #Prevents zoom
+                             fixedrange=True, #Prevents zoom
+                             automargin=True), #Prevents axis title from overlapping axis
```
Comment (radumas) [inline app.py:537]: Important!

File: app.py [hunk 9]
```diff
@@ -575,55 +582,88 @@ def generate_figure(street, direction, day_type='Weekday', period='AMPK',
...
+                            html.Div([html.B('Travel Time', style={'background-color':'#E9A3C9'}),' 1+ min', html.B(' longer'), ' than baseline']),
+                            html.Div([html.B('Travel Time', style={'background-color':'#A1D76A'}),' 1+ min', html.B(' shorter'), ' than baseline']), 
+                                             
+                        ])
+        ])                    
+
+app.layout = html.Div([
+            dbc.Row(
+                dbc.Col([html.H1(children=TITLE, id='title')],
+                                width=12, style={'backgroundColor': "#165788"})
```
Comment (radumas) [inline app.py:626]: If this is set here, why is it also set in `dashboard.css`?

File: app.py [hunk 10]
```diff
@@ -575,55 +582,88 @@ def generate_figure(street, direction, day_type='Weekday', period='AMPK',
...
+                                                html.Div(id = GRAPHDIVS[0], children=dcc.Graph(id=GRAPHS[0])),
+                                                html.H2(id=STREETNAME_DIV[1], style={'fontSize':20}),
+                                                html.Div(id = GRAPHDIVS[1], children=dcc.Graph(id=GRAPHS[1])),
+                            html.Div(children=[
+                                    html.Div(className="box_baseline", style={'display':'inline-block', 'margin-left':"30px"}),
+                                    html.Span("Baseline  ", style={'display':'inline-block', 'margin-left':"10px"}),
+                                    html.Div(className="box_closure", style={'display':'inline-block', 'margin-left':"20px"}),
+                                    html.Span("Closure  ", style={'display':'inline-block', 'margin-left':"10px"}), 
+                                    html.Div(className="box_selected", style={'display':'inline-block', 'margin-left':"20px"}),
+                                    html.Span("Selected  ", style={'display':'inline-block', 'margin-left':"10px"}),
```
Comment (radumas) [inline app.py]: This should be labelled as the legend.
Maybe create it above and drop it in here as a variable?

File: app.py [hunk 11]
```diff
@@ -868,7 +918,8 @@ def update_street_name(*args):
 
 [create_update_street_name(i) for i in [0,1]]
 
-@app.callback(Output(STREET_TITLE, 'children'),
+@app.callback([Output(STREET_TITLE, 'children'),
+                Output(TIME_TITLE, 'children')],
```
Comment (radumas) [inline app.py:922]: Innovation!

**Review by radumas — CHANGES_REQUESTED** [review]
Some questions inline. 
The only two that are really blocking merger into the live dashboard:
- [x] Should the fonts be Libre Franklin?
- [ ] Do I need to change anything in the heroku postgresql database?

There are other question/comments but they don't affect the appearance of the Dash, so could be dealt with later.

File: assets/dashboard.css (hunk 2 above)
Comment (chmnata) [reply assets/dashboard.css]: or Franklin Gothic?

File: app.py (hunk 5 above)
Comment (chmnata) [reply app.py:53]: Yes, `data_analysis.richmond_closure_week` and `data_analysis.richmond_closure_month` were changed. They are in [sql/month_week.sql](https://github.com/CityofToronto/bdit_traffic_dashboard/blob/richmond-dev/sql/month_week.sql).

Comment (chmnata) [reply app.py:53]: I also changed `data_analysis.richmond_dash_baseline`, in sql/baseline_view.sql

**Comment by radumas** [comment]
Creating tables for weeks and months to port over to Heroku

```sql
SELECT * INTO data_analysis.richmond_closure_months_tbl
FROM data_analysis.richmond_closure_months
LIMIT 1;

SELECT * INTO data_analysis.richmond_closure_weeks_tbl
FROM data_analysis.richmond_closure_weeks
LIMIT 1;
```

Transferring the schema to heroku
```bash
pg_dump -d bigdata -h 10.160.12.47 -t 'data_analysis.richmond_closure_*_tbl' -F p -O --schema-only | psql $heroku_string
```

Change table names:
```sql
ALTER TABLE data_analysis.richmond_closure_weeks_tbl RENAME TO richmond_closure_weeks; ALTER TABLE data_analysis.richmond_closure_months_tbl RENAME TO richmond_closure_months;
```

Sync data

```bash
psql $heroku_rds -c "\COPY (SELECT month, month_number FROM data_analysis.richmond_closure_months) TO STDOUT;" | psql $heroku_string -c "TRUNCATE data_analysis.richmond_closure_months; COPY data_analysis.richmond_closure_months(month, month_number) FROM STDIN;"
psql $heroku_rds -c "\COPY (SELECT week, week_number FROM data_analysis.richmond_closure_weeks) TO STDOUT;" | psql $heroku_string -c "TRUNCATE data_analysis.richmond_closure_weeks; COPY data_analysis.richmond_closure_weeks(week, week_number) FROM STDIN;"
```

About to include DAG

File: sql/baseline_view.sql [hunk 12]
```diff
@@ -7,23 +7,7 @@ CREATE OR REPLACE VIEW data_analysis.richmond_dash_baseline AS
...
-                END AS from_intersection,
-                CASE
-                    WHEN (a.street = ANY (ARRAY['Adelaide'::text, 'Front'::text, 'King'::text, 'Adelaide'::text, 'Richmond'::text, 'Queen'::text, 'Dundas'::text])) AND a.direction = 'EB'::text THEN 'Yonge'::text
-                    WHEN (a.street = ANY (ARRAY['Bathurst'::text, 'Spadina'::text, 'University'::text])) AND a.direction = 'NB'::text THEN 'Dundas'::text
-                    WHEN (a.street = ANY (ARRAY['Adelaide'::text, 'Front'::text, 'King'::text, 'Adelaide'::text, 'Richmond'::text, 'Queen'::text, 'Dundas'::text])) AND a.direction = 'WB'::text THEN 'Bathurst'::text
-                    WHEN (a.street = ANY (ARRAY['Bathurst'::text, 'Spadina'::text, 'University'::text])) AND a.direction = 'SB'::text THEN 'Front'::text
-                    WHEN a.street = 'Wellington'::text THEN 'Blue Jays'::text
-                    ELSE NULL::text
-                END AS to_intersection,
+         SELECT a.street, from_intersection, to_intersection, street_suffix,
```
Comment (radumas) [inline sql/baseline_view.sql:10]: Much cleaner

**Review by radumas — APPROVED** [review]

**Pull request merged**